zerver/models.py

```diff
@@ -1854,3 +1855,13 @@ def get_bot_services(user_profile_id):
 def get_service_profile(user_profile_id, service_name):
     # type: (str, str) -> Service
     return Service.objects.get(user_profile__id=user_profile_id, name=service_name)
+
+def get_service_dicts_for_bots(user_profile_id):
+    # type: (int) -> List[Dict[str, Any]]
+    services = get_bot_services(str(user_profile_id))
+    service_dicts = ([{'name': service.name,
+                       'base_url': service.base_url,
+                       'interface': service.interface,
+                       }
+                      for service in services])
+    return service_dicts
```

Comment:
Hrm.  This `models.py` logic is going to result in us doing database queries in a loop calling `get_bot_services`.  Fixing this will be somewhat complicated; I can try to help with that.